File: sample/UseMaven/GNAdSamples/javaSample/src/main/java/jp/co/geniee/samples/common/ImageGetTask.java
package jp.co.geniee.samples.common;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.AsyncTask;
import android.widget.ImageView;

import java.io.IOException;
import java.io.InputStream;
import java.net.URL;

public class ImageGetTask extends AsyncTask<Void, Void, Bitmap> {
    private String url;
    private final ThreadLocal<ImageView> imageView = new ThreadLocal<>();
    private String tag = "";

    public ImageGetTask(ImageView imageView, String url) {
        this.imageView.set(imageView);
        this.url = url;
        if (imageView.getTag() != null) {
            tag = imageView.getTag().toString();
        }
    }

    @Override
    protected Bitmap doInBackground(Void... params) {
        Bitmap bitmap;
        try {
            URL imageUrl = new URL(url);
            InputStream input;
            input = imageUrl.openStream();
            bitmap = BitmapFactory.decodeStream(input);
            input.close();
            return bitmap;
        } catch (IOException e) {
            return null;
        }
    }

    @Override
    protected void onPostExecute(Bitmap bitmap) {
        if (tag.equals(imageView.get().getTag())) {
            imageView.get().setImageBitmap(bitmap);
        }
    }
}

File: sample/kotlinSample/src/main/java/jp/co/geniee/samples/gnad/common/ImageGetTask.kt
package jp.co.geniee.samples.gnad.common

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.os.AsyncTask
import android.widget.ImageView
import java.io.IOException
import java.io.InputStream
import java.net.URL

class ImageGetTask(imageView: ImageView, private val url: String) : AsyncTask<Void, Void, Bitmap>() {
    private val imageView = ThreadLocal<ImageView>()
    private var tag = ""

    init {
        this.imageView.set(imageView)
        if (imageView.tag != null) {
            tag = imageView.tag.toString()
        }
    }

    override fun doInBackground(vararg params: Void): Bitmap? {
        val bitmap: Bitmap
        return try {
            val imageUrl = URL(url)
            val input: InputStream
            input = imageUrl.openStream()
            bitmap = BitmapFactory.decodeStream(input)
            input.close()
            bitmap
        } catch (e: IOException) {
            null
        }

    }

    override fun onPostExecute(bitmap: Bitmap?) {
        if (tag == imageView.get()!!.tag) {
            imageView.get()!!.setImageBitmap(bitmap)
        }
    }
}
